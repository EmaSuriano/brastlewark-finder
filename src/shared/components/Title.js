import styled from 'styled-components';
import PropTypes from 'prop-types';

const Title = styled.h1`
  color: ${props => props.theme[props.color]};
  font-family: 'Slabo 27px', serif;
  text-align: center;
  font-size: ${props => props.fontSize}em;
`;

Title.propTypes = {
  theme: PropTypes.object.isRequired,
  fontSize: PropTypes.string,
  color: PropTypes.string,
};

Title.defaultProps = {
  fontSize: '7',
  color: 'light',
};

export default Title;
