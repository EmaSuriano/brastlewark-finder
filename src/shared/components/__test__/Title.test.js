import React from 'react';
import { shallow } from 'enzyme';
import Title from '../Title';

describe('<Title />', () => {
  it('should render Title with defaultValues', () => {
    expect(shallow(<Title />)).toMatchSnapshot();
  });

  it('should render Title with red color', () => {
    expect(shallow(<Title color="red" />)).toMatchSnapshot();
  });

  it('should render Title with fontSize', () => {
    expect(shallow(<Title fontSize="2" />)).toMatchSnapshot();
  });
});
