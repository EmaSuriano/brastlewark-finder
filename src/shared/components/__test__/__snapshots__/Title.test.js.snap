// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render Title with defaultValues 1`] = `
.c0 {
  font-family: 'Slabo 27px',serif;
  text-align: center;
  font-size: 7em;
}

<h1
  className="c0"
  color="light"
  fontSize="7"
/>
`;

exports[`<Title /> should render Title with fontSize 1`] = `
.c0 {
  font-family: 'Slabo 27px',serif;
  text-align: center;
  font-size: 2em;
}

<h1
  className="c0"
  color="light"
  fontSize="2"
/>
`;

exports[`<Title /> should render Title with red color 1`] = `
.c0 {
  font-family: 'Slabo 27px',serif;
  text-align: center;
  font-size: 7em;
}

<h1
  className="c0"
  color="red"
  fontSize="7"
/>
`;
